### ISSUE TITLE
Option of doing optimizer.zero_grad(set_to_none=True). 

### ISSUE THREAD

**Issue by MarcoForte** [PyTorchLightning/pytorch-lightning#6534]
## 🚀 Feature

An option to set gradients to None instead of zero. This was introduced in PyTorch 1.7. [Documentation](https://pytorch.org/docs/master/optim.html#torch.optim.Optimizer.zero_grad).
### Motivation
Training Speed improvements, as discussed by PyTorch [here](https://pytorch.org/tutorials/recipes/recipes/tuning_guide.html#use-parameter-grad-none-instead-of-model-zero-grad-or-optimizer-zero-grad). 
### Pitch

I changed Line 1400 of https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/core/lightning.py. 
From 
```    
def optimizer_zero_grad(self, epoch: int, batch_idx: int, optimizer: Optimizer, optimizer_idx: int):  
    optimizer.zero_grad()
```
To 
```    
def optimizer_zero_grad(self, epoch: int, batch_idx: int, optimizer: Optimizer, optimizer_idx: int):
     optimizer.zero_grad(set_to_none=True)
```
I achieved a modest speedup in my training. In my case of 4096 image epoch, batchsize 32.   
My times before the change were **41.42s**, and after were **41.05s**. Measured as the min over three runs.

**Comment by rahumble** [comment]
@tchaton, how should this be done when `accumulate_grad_batches` in the Trainer is not 1? I'm not sure if there are other trainer args that are also incompatible with this method.

If I have
```
trainer = pl.Trainer(accumulate_grad_batches=2, ....)
```
and
```
def optimizer_zero_grad(self, epoch, batch_idx, optimizer, optimizer_idx):
    optimizer.zero_grad(set_to_none=True)
```
I get
```pytorch_lightning.utilities.exceptions.MisconfigurationException: When overriding `LightningModule` optimizer_step or optimizer_zero_grad, `accumulate_grad_batches` in `Trainer` should be 1. It ensures optimizer_step or optimizer_zero_grad are called on every batch.```
